tests/test_2022_05_31.py

```diff
@@ -0,0 +1,164 @@
+"""
+Tests for voting 31/05/2022.
+"""
+import pytest
+
+from brownie import interface, chain, ZERO_ADDRESS
+
+from scripts.vote_2022_05_31 import (self_owned_burn_role_params, start_vote,
+     get_burn_role_old_owner)
+from tx_tracing_helpers import *
+from utils.config import (contracts, lido_dao_steth_address,
+    network_name, lido_dao_voting_address, 
+    lido_dao_composite_post_rebase_beacon_receiver, lido_dao_self_owned_steth_burner)
+from event_validators.permission import (Permission, PermissionP, validate_permission_create_event,
+    validate_permission_revoke_event, validate_permission_grantp_event)
+from event_validators.lido import validate_set_el_rewards_vault_withdrawal_limit_event
+from event_validators.oracle import validate_beacon_report_receiver_set_event
+from event_validators.composite_post_rebase_beacon_receiver import validate_composite_receiver_callback_added_event
+from utils.permission_parameters import encode_permission_params
+
+
+# RESUME_ROLE
+permission_resume_role = Permission(
+    entity=lido_dao_voting_address,
+    app=lido_dao_steth_address,  # Lido
+    role='0x2fc10cc8ae19568712f7a176fb4978616a610650813c9d05326c34abb62749c7')
+
+# STAKING_PAUSE_ROLE 
+permission_staking_pause_role = Permission(
+    entity=lido_dao_voting_address,
+    app=lido_dao_steth_address,  # Lido
+    role='0x84ea57490227bc2be925c684e2a367071d69890b629590198f4125a018eb1de8')
+
+# SET_EL_REWARDS_WITHDRAWAL_LIMIT_ROLE
+permission_elrewards_set_limit_role = Permission(
+    entity=lido_dao_voting_address,
+    app=lido_dao_steth_address,  # Lido
+    role='0xca7d176c2da2028ed06be7e3b9457e6419ae0744dc311989e9b29f6a1ceb1003')
+
+# BURN_ROLE on Voting
+permission_burn_on_voting = Permission(
+    entity=get_burn_role_old_owner(),
+    app=lido_dao_steth_address,  # Lido
+    role='0xe97b137254058bd94f28d2f3eb79e2d34074ffb488d042e3bc958e0a57d2fa22')
+
+# BURN_ROLE on SelfOwnedStETHBurner
+permission_burn_on_steth_burner = PermissionP(
+    entity=lido_dao_self_owned_steth_burner,
+    app=lido_dao_steth_address,  # Lido
+    role='0xe97b137254058bd94f28d2f3eb79e2d34074ffb488d042e3bc958e0a57d2fa22',
+    params=encode_permission_params(self_owned_burn_role_params()))
```

Comment:
Looks like we need a lot of acl tests here. Especially for params.

Reply:
Agree, I am preparing them using the dedicated file

Reply:
added function check in `validate_parametrized_burn_permissions()`, this?